WordPress/Classes/Utility/WPUserAgent.m

```diff
@@ -35,17 +35,33 @@ - (void)setupUserAgent
     [[NSUserDefaults standardUserDefaults] registerDefaults:dictionary];
 }
 
-#pragma mark - WordPress User-Agent
+#pragma mark - Default and WordPress User-Agents
+
+- (NSString *)defaultUserAgent
+{
+    // Temporarily unset "UserAgent" from registered user defaults so that we
+    // always get the default value, independently from what's currently set as
+    // User-Agent
+    NSDictionary *originalRegisteredDefaults = [[NSUserDefaults standardUserDefaults] volatileDomainForName:NSRegistrationDomain];
+
+    NSMutableDictionary *tempRegisteredDefaults = [NSMutableDictionary dictionaryWithDictionary:originalRegisteredDefaults];
+    [tempRegisteredDefaults removeObjectForKey:WPUserAgentKeyUserAgent];
+    [[NSUserDefaults standardUserDefaults] registerDefaults:tempRegisteredDefaults];
+    
+    NSString *userAgent = [[[UIWebView alloc] init] stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+    NSAssert(userAgent != nil, @"User agent shouldn't be nil");
+    NSAssert(! [userAgent isEmpty], @"User agent shouldn't be empty");
+    
+    [[NSUserDefaults standardUserDefaults] registerDefaults:originalRegisteredDefaults];
+    
+    return userAgent;
+}
```

Comment:
> It is, however I've assumed that WPUserAgent is only to be used with UIWebView which always runs on the main thread (as the rest of the UIKit's classes). Said class exposes useDefaultUserAgent and useWordPressUserAgent which only have effect on UIWebView and not on other means of using HTTP.

Some parts of the code use `currentUserAgent` though to get whatever's set in NSUserDefaults at the moment, but I don't think temporarily resetting NSUserDefaults could introduce new problems here.